modules/social_features/social_user_export/src/Plugin/UserExportPluginInterface.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\social_user_export\Plugin;
+
+use Drupal\Component\Plugin\PluginInspectionInterface;
+use Drupal\profile\Entity\ProfileInterface;
+use Drupal\user\UserInterface;
+
+/**
+ * Defines an interface for User export plugin plugins.
+ */
+interface UserExportPluginInterface extends PluginInspectionInterface {
+
+  /**
+   * Returns the header.
+   *
+   * @return \Drupal\Core\StringTranslation\TranslatableMarkup|string
+   *   The header.
+   */
+  public function getHeader();
+
+  /**
+   * Returns the value.
+   *
+   * @param \Drupal\user\UserInterface $entity
+   *   The User entity to get the value from.
+   *
+   * @return string
+   *   The value.
+   */
+  public function getValue(UserInterface $entity);
+
+  /**
+   * Get the Profile entity.
+   *
+   * @param \Drupal\user\UserInterface $entity
+   *   The user entity to get the profile from.
+   *
+   * @return \Drupal\profile\Entity\ProfileInterface|null
+   *   Returns the Profile or NULL if profile does not exist.
+   */
+  public function getProfile(UserInterface $entity);
```

Comment:
The only two functions that we **really need** are `getHeader()` and `getValue()`. I don't think we have to define the other functions like `getProfile()` and `profileGetFieldValue()` in the Interface. We cannot let them live as additional functions in the pluginBase for whichever plugin wants to extend those.

@jochemvn check?

Reply:
Not sure what you are trying to say here @bramtenhove . You mean we should only put functions in the Interface which are really needed?